src/foundry/common/abstract/type-data.d.mts

```diff
@@ -16,10 +19,9 @@ declare const _InternalTypeDataModelConst: _InternalTypeDataModelInterface;
 declare class _InternalTypeDataModel<
   Schema extends DataSchema,
   Parent extends Document<DataSchema, any, any>,
-  BaseData extends Record<string, unknown> = Record<string, never>,
```

Comment:
The removal of `BaseData` is not correct. These are two different phases with two different possibilities for what's set here.

These different phases can be important in the context of initialization order etc.